### ISSUE TITLE
Some inference-only serialized Inductor SDPA patterns?

### ISSUE THREAD

**Issue by sanchitintel** [pytorch/pytorch#122119]
## Background

Currently, some [SDPA patterns in Inductor](https://github.com/pytorch/pytorch/blob/main/torch/_inductor/fx_passes/fuse_attention.py) have inference-only UTs ([`check_train` is `False` for them](https://github.com/pytorch/pytorch/blob/main/test/inductor/test_fused_attention.py#L520)), but they are also being serialized for training.

There are two possibilities:
1. UTs of such patterns are not representative of the corresponding training pattern, OR
2. The above case, and also, these training patterns are not matched even when their representative/reference workload (not UT, but end-to-end model) is run, and only their inference counterpart matches.

## Proposal

In case of possibility 1 (UTs of such patterns not being representative of the corresponding training pattern) discussed above, those UTs should be fixed.

But in case of possibility 2 (the training patterns don't even match when the reference workload they were created for is run), would it be better not to serialize them until the issues that cause mismatch are resolved?

### Benefits
**Performance-impact**: insignificant improvement in compile-time.
**Main benefit**: Code-clarity. Currently, it's not obvious from looking at `torch/_inductor/fx_passes/fuse_attention.py` that some training patterns don't match when their representative workload is run.

Thanks!

### Alternatives

We could simply add comments if such training patterns are to be retained although they don't match when their representative workload is run.

**Comment by sanchitintel** [comment]
Hi @eellison, please advise if it'd be okay to resolve this issue in a separate PR after #121866, or if I should address it in the same PR. Thanks!
